### PR TITLE
docs(stories): Add a story for the <Confirm> component

### PR DESCRIPTION
**Pull request by ryan953**
SCR-20231207-kxfp

### CODE REVIEW

File: static/app/components/confirm.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+export default storyBook(Confirm, story => {
+  story('Triggers', () => (
+    <Fragment>
+      <p>
+        There are two way to use <JSXNode name="Confirm" />, either as a wrapper around a
+        trigger, or by calling <code>openConfirmModal()</code> in a callback.
+      </p>
+      <p>
+        I recommend calling <code>openConfirmModal()</code>.
```
Comment (michellewzhang) [inline static/app/components/confirm.stories.tsx:24]: Should this say "we recommend"...? 🤔

File: static/app/components/confirm.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        <p>Button clicks: {clicks} </p>
+        <Matrix
+          propMatrix={{
+            disabled: [false, true],
+            stopPropagation: [false, true],
+          }}
+          render={props => (
+            <Button onClick={() => setClicks(prev => prev + 1)}>
+              <Confirm {...props}>
+                <Green>Open modal</Green>
```
Comment (michellewzhang) [inline static/app/components/confirm.stories.tsx:172]: nit: could we make the `<Green>` bit span the entire button? I noticed that if you click outside the green (on the little white bits on either side) when `stopPropagation = true`,` it'll still up the button click count

**Review by michellewzhang — APPROVED** [review]
disabled auto merge in case you want to add changes but feel free to enable again! just nits

File: static/app/components/confirm.stories.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/components/confirm.stories.tsx:172]: Ya, that was part of the demo. I can do something to make it more explicit.

**Pull request merged**